ivy/functional/ivy/general.py

```diff
@@ -2560,6 +2562,21 @@ def shape(
     return current_backend(x).shape(x, as_array)
 
 
+def set_shape_array_mode() -> None:
+    global shape_array_mode
```

Comment:
this should use a stack rather than a single global value enabling us to use nested blocks of setting and unsetting. See other global variables for examples.